src/onlyUpdateForProps.js

```diff
@@ -0,0 +1,8 @@
+import onlyUpdateForKeys from 'onlyUpdateForKeys';
+
+const onlyUpdateForProps = (BaseComponent) => {
+  const propKeys = Object.keys(BaseComponent.propTypes || {});
```

Comment:
That's right. Maybe we should just return it untouched, e.g `BaseComponent` + optionally print a `warning` since in this case it's clearly an implementation issue.